nautobot/utilities/forms/fields.py

```diff
@@ -669,12 +669,17 @@ def to_python(self, value):
         return super().to_python(value)
 
 
-class SlugOrPKMultipleChoiceField(django_filters.fields.ModelMultipleChoiceField):
+class MultiMatchModelMultipleChoiceField(django_filters.fields.ModelMultipleChoiceField):
     """
-    Field to support matching an object's `pk` or `slug` field depending on the value
-    supplied. Raises ValidationError if neither field matches.
+    Filter field to support matching on the PK or supplied `natural_key` fields. `natural_key`
+    defaults to `slug` if not provided. Raises ValidationError if none of the fields match the
+    requested value.
     """
 
+    def __init__(self, *args, **kwargs):
+        self.natural_key = kwargs.pop("natural_key", "slug")
```

Comment:
A next step (which I may pursue as part of #1891) would be to automatically set `field_name` and/or `to_field_name` accordingly.